### PR TITLE
LLDP - support for basic TLVs

### PR DESCRIPTION
**Pull request by speakinghedge**
- based on IEEE 802.1AB-2016
- adds support for basic LLDP TLVs
- offers strict mode (enforces order of TLVs, multiplicity of TLVs, string lengths)

### CODE REVIEW

**Comment by gpotter2** [comment]
Please follow the python 3 compatibility rules:
https://github.com/secdev/scapy/blob/master/CONTRIBUTING.md#python-2-and-3-compatibility

File: scapy/contrib/lldp.py [hunk 1]
```diff
@@ -0,0 +1,714 @@
...
+        0x00: 'end of LLDPDU',
+        0x01: 'chassis id',
+        0x02: 'port id',
+        0x03: 'time to live',
+        0x04: 'port description',
+        0x05: 'system name',
+        0x06: 'system description',
+        0x07: 'system capabilities',
+        0x08: 'management address',
+        xrange(0x09, 0x7e): 'reserved - future standardization',
```
Comment (gpotter2) [inline scapy/contrib/lldp.py]: As specified in CONTRIBUTING.md, please use `range` from `scapy.modules.six.moves`

File: scapy/contrib/lldp.py [hunk 2]
```diff
@@ -0,0 +1,714 @@
...
+    LLDP_CHASSIS_ID_TLV_SUBTYPES = {
+        0x00: 'reserved',
+        0x01: 'chassis component',
+        0x02: 'interface alias',
+        0x03: 'port component',
+        0x04: 'MAC address',
+        0x05: 'network address',
+        0x06: 'interface name',
+        0x07: 'locally assigned',
+        xrange(0x08, 0xff): 'reserved'
```
Comment (gpotter2) [inline scapy/contrib/lldp.py]: Same here

File: scapy/contrib/lldp.py [hunk 3]
```diff
@@ -0,0 +1,714 @@
...
+    LLDP_PORT_ID_TLV_SUBTYPES = {
+        0x00: 'reserved',
+        0x01: 'interface alias',
+        0x02: 'port component',
+        0x03: 'MAC address',
+        0x04: 'network address',
+        0x05: 'interface name',
+        0x06: 'agent circuit ID',
+        0x07: 'locally assigned',
+        xrange(0x08, 0xff): 'reserved'
```
Comment (gpotter2) [inline scapy/contrib/lldp.py]: Same on all xrange

**Comment by gpotter2** [comment]
Could you also add
`ETHER_TYPES['LLDP'] = LLDP_ETHER_TYPE`
when loading ? That would update ETHER_TYPES in `l2.py`

**Comment by speakinghedge** [comment]
Thank you for the instant feedback. Hope I was able to address all issues.

**Comment by gpotter2** [comment]
> Thank you for the instant feedback.

I really was lucky on that part :)

Looks better, thx.

File: scapy/contrib/lldp.uts [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+% LLDP test campaign
+
+#
+# execute test:
+# > test/run_tests -P "load_contrib('lldp')" -t scapy/contrib/lldp.uts
+#
+
++ Syntax check
+= Import the LLDP layer
+from scapy.contrib.lldp import *
```
Comment (guedou) [inline scapy/contrib/lldp.uts]: This is not needed. Unit tests should be launched as:
`./test/run_tests -P "load_contrib('gtp')" ../scapy/contrib/lldp.uts`

**Comment by p-l-** [comment]
Hi,

Thanks for your contribution!

An implementation exists in `scapy/layers/lldp.py`. Have you seen it?

File: scapy/contrib/lldp.uts (hunk 4 above)
Comment (speakinghedge) [inline scapy/contrib/lldp.uts]: I copied this from other tests cause I was under the impression that this import test is done to check if the import fails cause something is wrong with the module path or with the modules syntax..

result of the first test keeping the include in the .uts (just added a X inside the lldp.py):
```
###(000)=[failed] Import the LLDP layer

>>> from scapy.contrib.lldp import *
Traceback (most recent call last):
  File "<input>", line 2, in <module>
  File "/home/hecke/projects/private/scapy/scapy/contrib/lldp.py", line 44, in <module>
    X
NameError: name 'X' is not defined
```
import test removed leads to regressions in functional tests:
```
###(000)=[failed] build basic LLDP frames

>>> frm = Ether(src='01:01:01:01:01:01', dst=LLDP_NEAREST_BRIDGE_MAC)/ \
...       LLDPDUChassisID(subtype=LLDPDUChassisID.SUBTYPE_MAC_ADDRESS, id=b'\x06\x05\x04\x03\x02\x01') / \
.......
...       LLDPDUEndOfLLDPDU()
Traceback (most recent call last):
  File "<input>", line 13, in <module>
NameError: name 'LLDP_NEAREST_BRIDGE_MAC' is not defined
```

so keeping the include test would give an early and clean indication that something is wrong with the module.

removed import [here](https://github.com/secdev/scapy/pull/701/commits/cf92af670c950b461c5c13cfd9ddad756a4ead8d) to follow the convention of the project.

**Comment by speakinghedge** [comment]
>> An implementation exists in scapy/layers/lldp.py. Have you seen it?

No. On which branch?

**Comment by p-l-** [comment]
Sorry, that's in [scapy-com](https://bitbucket.org/secdev/scapy-com/src/cc06add6dbd9b470b6d2696507989574910df811/scapy/layers/lldp.py?at=default&fileviewer=file-view-default) and has never landed in Scapy. I have adapted it but that's... only on my computer now.

**Pull request merged**